src/solver/writer/zip_writer.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <mutex>
+#include <string>
+
+#include <antares/benchmarking.h> // Timer
+
+#include <yuni/job/queue/service.h>
+#include <yuni/job/job.h>
+#include <yuni/core/string.h>
+
+#include "i_writer.h"
+
+namespace Antares
+{
+namespace Solver
+{
+class ZipWriter;
+template<class ContentT>
+class ZipWriteJob final : public Yuni::Job::IJob
+{
+public:
+    ZipWriteJob(ZipWriter& writer,
+                const std::string& entryPath,
+                ContentT& content,
+                Benchmarking::IDurationCollector* duration_collector);
+    virtual void onExecute() override;
+
+private:
+    // Pointer to Zip handle
+    void* pZipHandle;
+    // Protect pZipHandle against concurrent writes, since minizip-ng isn't thread-safe
+    std::mutex& pZipMutex;
+    // Entry path for the new file within the zip archive
+    const std::string pEntryPath;
+    // Content of the new file
+    ContentT pContent;
+    // Benchmarking. How long do we wait ? How long does the zip write take ?
+    Benchmarking::IDurationCollector* pDurationCollector;
+};
+
+class ZipWriter : public IResultWriter
+{
+public:
+    ZipWriter(std::shared_ptr<Yuni::Job::QueueService> qs,
+              const char* archivePath,
+              Benchmarking::IDurationCollector* duration_collector);
+    ~ZipWriter();
+    void addJob(const std::string& entryPath, Yuni::Clob& entryContent) override;
+    void addJob(const std::string& entryPath, std::string& entryContent) override;
+    void addJob(const std::string& entryPath, Antares::IniFile& entryContent) override;
```

Comment:
> Why don't we execute the work of adding the zip archive a piece of data in the same thread than the calling thread ?

Because zip files can't be written concurrently, _ie_ since the lib isn't thread-safe. All operations on it are `mutex`ed. If we wrote results immediately in this case, a lot of CPU time would be wasted waiting for write operations to be completed. We don't want that, so we give the responsability of writing data to a separate thread.